src/hotspot/share/services/heapDumperCompression.cpp

```diff
@@ -237,17 +237,14 @@ CompressionBackend::~CompressionBackend() {
   delete _lock;
 }
 
-void CompressionBackend::flush_buffer() {
-  assert(_active, "Must be active");
-
-  MonitorLocker ml(_lock, Mutex::_no_safepoint_check_flag);
+void CompressionBackend::flush_buffer_without_lock(MonitorLocker* ml) {
```

Comment:
I'm trying to think of a better name since "without_lock" makes it sound like there is no locking, but there is. The difference is the lock is passed in. How about just calling it `flush_buffer(MonitorLocker* ml)` and rely on C++ method overloading to disambiguate it from `flush_buffer()`.